File: Week 01/id_401/26.remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    if(!nums || nums.length<=1) return nums.length;
    let temp=nums[0];
    let i=1;
    while(i<nums.length){
        if(temp===nums[i]){
            nums.splice(i,1);
        }
        else{
            temp=nums[i];
            i++;
        }
    }
    return nums.length;
};

File: Week 01/id_586/26-remove-duplicates-from-sorted-array.c
/* 双指针法:
 *  两个指针 ii 和 jj，其中 ii 是慢指针，而 jj 是快指针。只要 nums[i] = nums[j]nums[i]=nums[j]，我们就增加 jj 以跳过重复项
 */
int removeDuplicates(int* nums, int numsSize){
    if (numsSize <= 0) return 0;
    
    int i = 0;
    int j = 0;
    
    for (j = 1; j < numsSize; j++) {
        if (nums[i] != nums[j]) {
            i++;
            nums[i] = nums[j];
        }
    }
    
    return i + 1;
}
